tensorbay/label/tests/test_subcatalog.py

```diff
@@ -0,0 +1,203 @@
+from .. import (
+    Box2DSubcatalog,
+    Box3DSubcatalog,
+    ClassificationSubcatalog,
+    Keypoints2DSubcatalog,
+    Polygon2DSubcatalog,
+    Polyline2DSubcatalog,
+    SentenceSubcatalog,
+)
+from ..attributes import AttributeInfo
+from ..supports import (
+    CategoryInfo,
+    KeypointsInfo,
+    SupportAttributes,
+    SupportCategories,
+    SupportIsTracking,
+)
+
+_DATA_CATEGORIES = {"categories": [{"name": "0"}, {"name": "1"}]}
+_DATA_ATTRIBUTES = {"attributes": [{"name": "gender", "enum": ["male", "female"]}]}
+_DATA_DELIMITER = {"categoryDelimiter": "-"}
+_DATA_TRACKING = {"isTracking": True}
+_DATA_NAMES = [
+    "L_shoulder",
+    "L_Elbow",
+    "L_wrist",
+    "R_Shoulder",
+    "R_Elbow",
+]
+_DATA_SKELETON = [(0, 1), (1, 2), (3, 4), (4, 5)]
+_DATA_KEYPOINTS = {
+    "keypoints": [
+        {
+            "number": 5,
+            "names": _DATA_NAMES,
+            "skeleton": _DATA_SKELETON,
+        }
+    ]
+}
+_LEXICON = ["齐", "q", "i2"]
+_DATA_SENTENCE = {"isSample": True, "sampleRate": 16000, "lexicon": [_LEXICON]}
+_COMMON_SUBCATALOG_CONTENT = dict(_DATA_TRACKING, **_DATA_CATEGORIES, **_DATA_ATTRIBUTES)
+_CLASSIFICATION_SUBCATALOG = dict(_DATA_CATEGORIES, **_DATA_ATTRIBUTES, **_DATA_DELIMITER)
+_KEYPOINTS2D_SUBCATALOG = dict(
+    _DATA_TRACKING, **_DATA_CATEGORIES, **_DATA_ATTRIBUTES, **_DATA_KEYPOINTS
+)
+_SENTENCE_SUBCATALOG = dict(_DATA_ATTRIBUTES, **_DATA_SENTENCE)
+
+
+@pytest.fixture
+def categories():
+    categories = NameOrderedDict()
+    categories.append(CategoryInfo("0"))
+    categories.append(CategoryInfo("1"))
+    return categories
+
+
+@pytest.fixture
+def attributes():
+    attributes = NameOrderedDict()
+    attributes.append(AttributeInfo("gender", enum=["male", "female"]))
+    return attributes
+
+
+@pytest.fixture
+def keypoints():
+    return KeypointsInfo(
+        5,
+        names=_DATA_NAMES,
+        skeleton=_DATA_SKELETON,
+    )
+
+
+class TestClassificationSubcatalog:
+    def test_init_subclass(self):
+        classification_subcatalog = ClassificationSubcatalog()
+        classification_subcatalog._supports = (SupportCategories, SupportAttributes)
+
+    def test_loads(self):
+        classification_subcatalog = ClassificationSubcatalog.loads(_CLASSIFICATION_SUBCATALOG)
+        assert list(classification_subcatalog.categories) == ["0", "1"]
+        assert classification_subcatalog.attributes["gender"].enum == ["male", "female"]
+
+    def test_dumps(self, categories, attributes):
+        classification_subcatalog = ClassificationSubcatalog()
+        classification_subcatalog.category_delimiter = "-"
+        classification_subcatalog.categories = categories
+        classification_subcatalog.attributes = attributes
+        assert classification_subcatalog.dumps() == _CLASSIFICATION_SUBCATALOG
+
+
+class TestCommnoSubcatalog:
+    """This calss used to test
+    Box2DSubcatalog,
+    Box3DSubcatalog,
+    Polygon2DSubcatalog,
+    Polyline2DSubcatalog
+    """
```

Comment:
"""This class used to test multiple classes.

 Box2DSubcatalog, Box3DSubcatalog,
 Polygon2DSubcatalog, Polyline2DSubcatalog.
"""